### PR TITLE
Add editor-{copy,cut}-current to copy/cut the selection or otherwise current line

### PR DESCRIPTION
**Pull request by barnybug**
Similar to editor-duplicate-current and models the behaviour of a few other text editors - if you haven't a selection Ctrl+C/X will copy/cut the current line, instead of doing nothing. I find this very handy for moving a single line of code around.

### CODE REVIEW

**Comment by shalabhc** [comment]
Could you please share which other editors have this behavior for the standard cut/copy bindings? It seems a bit strange to me, though I'm not against new bindings for these.

**Comment by barnybug** [comment]
Sublime text:
https://gist.github.com/eteanga/1736542
and komodo:
http://komodoide.com/blog/2013-01/komodo-tips-tricks/

**Comment by barnybug** [comment]
Visual studio (not been a user in a long time though so can't verify!)

**Comment by barnybug** [comment]
Netbeans too.

Atom and Eclipse both have plugins doing this; (Atom by default deletes (not cuts) a single character forward of the cursor on ctrl+x with no selection, somewhat bizarrely).

**Comment by shalabhc** [comment]
OK then, I vote [+0](https://www.python.org/dev/peps/pep-0010/). @nilnor has the final say, of course.

But please update the `osx` versions too :smiley:

**Comment by barnybug** [comment]
Better than a -0 at least :-)
Split the PR into two commits - one adding the commands, and second updating the key bindings (mac too).
Can drop the keybinding change if it isn't entirely agreeable.

**Comment by nilnor** [comment]
My thought is whether it makes sense to keep the original operations in this case, or whether it 's better to just update the original methods. @shalabhc, @kirbyfan64: any votes on that? :)

**Comment by kirbyfan64** [comment]
Well, you could always change the bodies of `selection\cut` and `selection\copy` to select the current line if there is no active selection, instead of just returning. The problem is just that `Selection` (rightfully) doesn't know anything about the current buffer and cursor...

Maybe the `copy` and `cut` operations on `Selection` could take an argument `default_selection` or something?

**Comment by shalabhc** [comment]
I don't think the `cut` and `copy` methods on `selection` should be changed at all.

However, I'm in favor of changing the existing 'editor-copy' and 'editor-cut' commands instead of adding the new '-current' variants. The new commands are so similar and provide a superset of the functionality - why would anyone need 'editor-cut' but not just use 'editor-cut-current'?

**Comment by nilnor** [comment]
Yes, sorry about being unclear, but what I was attempting to say was what @shalabhc formulated more clearly, that the existing commands would be replaced rather than having two new variants. Well, it seems we think along the same lines though, so unless @kirbyfan64 is vehemently opposed let's go with updating the existing commands instead.

**Comment by kirbyfan64** [comment]
Sure!

**Comment by barnybug** [comment]
Thanks guys - nice to know changes are carefully considered. I've made the suggested changes.

File: lib/howl/ui/editor.moon [hunk 1]
```diff
@@ -989,6 +1002,8 @@ for cmd_spec in *{
   { 'scroll-up', 'Scrolls one line up', 'scroll_up' }
   { 'scroll-down', 'Scrolls one line down', 'scroll_down' }
   { 'duplicate-current', 'Duplicates the selection or current line', 'duplicate_current' }
+  { 'cut', 'Cut the selection or current line to the clipboard', 'cut' }
```
Comment (nilnor) [inline lib/howl/ui/editor.moon]: Final nitpick - the descriptions should be fixed to be in the same form as the other descriptions, e.g. "Cuts the", "Copies the". After that we're good to go :)

**Comment by barnybug** [comment]
Wording tweaked.

**Comment by nilnor** [comment]
Looks great, thanks!

**Pull request merged**